### PR TITLE
Add ProteomIQon PeptideSpectrumMatching

### PR DESCRIPTION
**Pull request by ZimmerA**
Given raw an MS run in the mzLite format, this tool iterates across all MS/MS scans, determines precursor charge states and possible peptide spectrum matches using reimplementations of SEQUEST, Andromeda and XTandem.

### CODE REVIEW

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    <requirements>
+        <requirement type="package" version="@VERSION@">proteomiqon-peptidespectrummatching</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #import re
+        #set basename = $re.sub(r'[^\w ,.\-+]','_',$instrumentOutput.element_identifier)
+        #if $outputParamfile:
+            cat '$paramfile' >> '$out_paramfile' &&
+        #end if
+        ln -s $instrumentOutput '${basename}.mzlite' &&
```
Comment (bgruening) [inline tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:17]: single quote

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        ln -s $out_psm '${basename}.psm' &&
```
Comment (bgruening) [inline tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:18]: single quote

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                },
+                "MatchingIonTolerancePPM": ${Andromeda.MatchingIonTolerancePPM}
+            }
+        }
+        ]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+    <param name="instrumentOutput" type="data" format="sqlite" label="Instrument output" help="Specify mass spectrometry data you want to analyze."/>
+    <param name="peptideDB" type="data" format="sqlite" label="Peptide database" help="Specify the peptide data base."/>
```
Comment (bgruening) [inline tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:50]: Just for your information, if you know some of the metadata of such a table (https://github.com/galaxyproject/galaxy/blob/87cd4ad3c79b586e9d92c29320e4adbcd817f3f6/lib/galaxy/datatypes/binary.py#L1685) you can filter the inputs here to only display relevant sqlite datasets to the user.

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    <section name="ChargeStateDeterminationParams" title="Charge state determination parameters" >
+        <param name="ExpectedMinimalCharge" type="integer" value="2" label="Expected minimal charge" help="Specify the minimum peptide ion charge state to consider."/>
```
Comment (bgruening) [inline tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:52]: would it make sense to add min= max= to a few of the floats/integers here?

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        <param name="DeltaMinIntensity" type="float" value="0.3" label="Delta min intensity" help="Specify minimum relative intensity to the prior peak when iterating through peak lists while performing charge state determination."/>
+        <param name="NrOfRndSpectra" type="integer" value="10000" label="Number of random spectra" help="When performing charge state determination we compared measured envelopes to the distribution of randomly generated spectra. This parameter tunes the size of the generated data set."/>
+    </section> 
+        <param name="LookUpPPM" type="float" value="30" label="Lookup PPM" help="Specify the width of the search space when retrieving in silico generated peptide ions from the peptide data base. The width is calculated relative to the precursor mz in ppm."/> 
+    <param name="nTerminalSeries" type="select" optional="false" label="N-terminal series" help="Specify the types of N-terminal charged ion fragments included in the generation of in silico fragment spectra." multiple="true" display="checkboxes">
```
Comment (bgruening) [inline tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:60]: indentation is wrong here, please use 4 spaces

Comment (bgruening) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:60]: actually one line above

**Review by bgruening — COMMENTED** [review]
Nice, this is pretty good! just a few small comments.

Can we maybe also reduce the size of sample.mzlite to under 1MB?

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml (hunk 1 above)
Comment (ZimmerA) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:17]: It is single quotes for me?

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml (hunk 3 above)
Comment (ZimmerA) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:50]: Ahh, interesting! I will take a look at that for the next versions!

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml (hunk 1 above)
Comment (bgruening) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:17]: The first one as well :)

Comment (ZimmerA) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:17]: Ahh, I'm blind, sorry  😅

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml (hunk 3 above)
Comment (bgruening) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml:50]: especially the last example: https://docs.galaxyproject.org/en/latest/dev/schema.html#id57

File: tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml (hunk 4 above)
Comment (ZimmerA) [reply tools/proteomiqon_peptidespectrummatching/proteomiqon_peptidespectrummatching.xml]: Added

**Review by bgruening — APPROVED** [review]

**Pull request merged**